docs/getting-started/walk-through/advanced-hera-features.md

```diff
@@ -254,32 +275,50 @@ def func() -> Annotated[str, Artifact(name="hello-art", path="/tmp/hello_world.t
 ```
 
 The idea is to save the returned value in a file according to this schema:
-`/hera/outputs/parameters/<name>`
-`/hera/outputs/artifacts/<name>`
-for easy access in subsequent `Steps`/`Tasks`. The output is also exposed in the `outputs` section of the resulting yaml. 
+* `/hera/outputs/parameters/<name>`
+* `/hera/outputs/artifacts/<name>`
+
+The output is also exposed in the `outputs` section of the resulting yaml. 
+
+The item returned from the function can be of any serialisable Pydantic type (or basic Python type) and needs to be
+`Annotated` with an `Artifact` or `Parameter`. The `Parameter`/`Artifact`'s `name` will be taken and used for creating
+the path for saving the item. If the annotation is `Artifact` and it contains a `path`, we use that `path` to save the
+output. See below for examples of an output `Parameter` or `Artifact`:
 
-The item returned from the function can be of any serialisable Pydantic type. That arbitrary type needs to be 
-`Annotated` with an `Artifact` or `Parameter`. One of 
-these is needed because we have to know where to save it to file at the end. The `Parameter`/`Artifact`'s `name` will 
-be taken and used for creating the path for saving the item. If the annotation is `Artifact` and it contains a `path`, 
-we use that `path` to save the output.
+Parameter output:
 
 ```python
 @script()
-def func(...) -> Annotated[arbitrary_pydantic_type, OutputItem]:
-    return output
+def func(...) -> Annotated[arbitrary_pydantic_type, Parameter(name="my-output")]:
+    return output  # will be saved to /hera/outputs/parameters/my-output
+```
+
+Artifact output:
+
+```python
+@script()
+def func(...) -> Annotated[arbitrary_pydantic_type, Artifact(name="my-output")]:
+    return output  # will be saved to /hera/outputs/artifacts/my-output
 ```
 
-We also allow the return type to be a `Tuple` of arbitrary Pydantic types with `Parameter`/`Artifact` annotations.
+For multiple outputs, the return type should be a `Tuple` of arbitrary Pydantic types with individual
+`Parameter`/`Artifact` annotations, and you should return a tuple from the function matching these types:
```

Comment:
Not sure if the direct "you should return" is congruent with the rest of the text. Maybe "and the function should return a tuple matching these types" would be better?